ios/browser/api/password/brave_password_api.mm

```diff
@@ -0,0 +1,472 @@
+                dateCreated:(NSDate*)dateCreated
+               dateLastUsed:(NSDate*)dateLastUsed
+        datePasswordChanged:(NSDate*)datePasswordChanged
+            usernameElement:(NSString*)usernameElement
+              usernameValue:(NSString*)usernameValue
+            passwordElement:(NSString*)passwordElement
+              passwordValue:(NSString*)passwordValue
+            isBlockedByUser:(bool)isBlockedByUser
+                     scheme:(PasswordFormScheme)scheme {
+  if ((self = [super init])) {
+    [self setUrl:url];
+
+    [self setSignOnRealm:signOnRealm];
+
+    if (dateCreated) {
+      [self setDateCreated:dateCreated];
+    }
+
+    if (dateLastUsed) {
+      [self setDateLastUsed:dateLastUsed];
+    }
+
+    if (datePasswordChanged) {
+      [self setDatePasswordChanged:datePasswordChanged];
+    }
+    if (usernameElement) {
+      [self setUsernameElement:usernameElement];
+    }
+
+    if (usernameValue) {
+      [self setUsernameValue:usernameValue];
+    }
+
+    if (passwordElement) {
+      [self setPasswordElement:passwordElement];
+    }
+
+    if (passwordValue) {
+      [self setPasswordValue:passwordValue];
+    }
+
+    self.isBlockedByUser = isBlockedByUser;
+
+    password_form_scheme_ =
+        brave::ios::PasswordFormSchemeForPasswordFormDigest(scheme);
+  }
+
+  return self;
+}
+
+- (id)copyWithZone:(NSZone*)zone {
+  IOSPasswordForm* passwordFormCopy = [[[self class] allocWithZone:zone] init];
+
+  if (passwordFormCopy) {
+    passwordFormCopy.url = self.url;
+    passwordFormCopy.signOnRealm = self.signOnRealm;
+    passwordFormCopy.dateCreated = self.dateCreated;
+    passwordFormCopy.dateLastUsed = self.dateLastUsed;
+    passwordFormCopy.datePasswordChanged = self.datePasswordChanged;
+    passwordFormCopy.usernameElement = self.usernameElement;
+    passwordFormCopy.usernameValue = self.usernameValue;
+    passwordFormCopy.passwordElement = self.passwordElement;
+    passwordFormCopy.passwordValue = self.passwordValue;
+    passwordFormCopy.isBlockedByUser = self.isBlockedByUser;
+    passwordFormCopy.scheme = self.scheme;
+  }
+
+  return passwordFormCopy;
+}
+
+- (void)updatePasswordForm:(NSString*)usernameValue
+             passwordValue:(NSString*)passwordValue {
+  if ([usernameValue length] != 0) {
+    [self setUsernameValue:usernameValue];
+  }
+
+  if ([passwordValue length] != 0) {
+    [self setPasswordValue:passwordValue];
+  }
+}
+
+- (void)setUrl:(NSURL*)url {
+  gurl_ = net::GURLWithNSURL(url);
+}
+
+- (NSURL*)url {
+  return net::NSURLWithGURL(gurl_);
+}
+
+- (void)setSignOnRealm:(NSString*)signOnRealm {
+  if ([signOnRealm length] != 0) {
+    signon_realm_ = base::SysNSStringToUTF8(signOnRealm);
+  } else {
+    signon_realm_ = gurl_.spec();
+  }
+}
+
+- (NSString*)signOnRealm {
+  return base::SysUTF8ToNSString(signon_realm_);
+}
+
+- (void)setDateCreated:(NSDate*)dateCreated {
+  date_created_ = base::Time::FromNSDate(dateCreated);
```

Comment:
This DCHECK's `dateCreated` but you have the property marked as nullable, so you need to not call `base::Time::FromNSDate` if dateCreated is nil

Reply:
fix at https://github.com/brave/brave-core/pull/12360/commits/f1ed2d6425c64f0f9a7d7075c7af10d5d38b04c5